server/engagement/views.py

```diff
@@ -1,51 +1,146 @@
-from django.shortcuts import get_object_or_404
-from rest_framework import generics, status
+from rest_framework.views import APIView
 from rest_framework.response import Response
-from django.http import Http404
-from django.db import transaction
-from django.db.models import F
-from engagement.models import DiscordMessageStats
-from members.models import User
+from rest_framework import status
+from custom_auth.permissions import IsAdmin
 from members.permissions import IsApiKey
+from .models import AttendanceSession
+from django.utils import timezone
+from datetime import datetime
+from .serializers import AttendanceSessionSerializer, MemberSerializer
+from django.shortcuts import get_object_or_404
+from members.models import User
+from rest_framework import generics
 
-import logging
+def parse_date(x):
+    date = datetime.strptime(x, "%Y-%m-%dT%H:%M:%S.%fZ")
 
-logger = logging.getLogger(__name__)
+    if timezone.is_aware(date):
+        return date
+    return timezone.make_aware(date)
 
+class CreateAttendanceSession(APIView):
+    permission_classes = [IsAdmin|IsApiKey]
```

Comment:
Do we want to be able to create attendance sessions from the discord bot? I thought, as it says in our design doc, we'd have a frontend React app for that.